presto-orc/src/main/java/com/facebook/presto/orc/reader/DecimalStreamReader.java

```diff
@@ -152,6 +158,7 @@ private void assureVectorSize()
         if (nullVector.length < requiredVectorLength) {
             nullVector = new boolean[requiredVectorLength];
             scaleVector = new long[requiredVectorLength];
+            systemMemoryContext.setBytes(requiredVectorLength * (SIZE_OF_LONG + SIZE_OF_BYTE));
```

Comment:
`requiredVectorLength` is an int and the size of constants are bytes, so there is a chance of overflow here. Let's cast it to long here (other option is to make that variable a long, but I like the cast approach as it's explicitly saying to the reader that we are aware of the overflow issue)

```
systemMemoryContext.setBytes((long) requiredVectorLength * (SIZE_OF_LONG + SIZE_OF_BYTE));
```